python/lsst/multiprofit/fitting/fit_catalog.py

```diff
@@ -0,0 +1,118 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CatalogExposureABC", "ColumnInfo", "CatalogFitterConfig"]
+
+from abc import ABC, abstractmethod
+from collections.abc import Iterable
+
+import astropy.units as u
+import lsst.pex.config as pexConfig
+import pydantic
+
+from ..modeller import ModelFitConfig
+from ..utils import frozen_arbitrary_allowed_config
+
+
+class CatalogExposureABC(ABC):
+    """Interface for catalog-exposure pairs."""
+
+    # TODO: add get_exposure (with Any return type?)
+
+    @abstractmethod
+    def get_catalog(self) -> Iterable:
+        """Return a row-iterable catalog covering an exposure."""
+
+
+class ColumnInfo(pydantic.BaseModel):
+    """Metadata for a column in a catalog."""
+
+    model_config = frozen_arbitrary_allowed_config
+
+    dtype: str = pydantic.Field(title="Column data type name (numpy or otherwise)")
+    key: str = pydantic.Field(title="Column key (name)")
+    description: str = pydantic.Field("", title="Column description")
+    unit: u.UnitBase | None = pydantic.Field(None, title="Column unit (astropy)")
+
+
+class CatalogFitterConfig(pexConfig.Config):
+    """Configuration for generic MultiProFit fitting tasks."""
+
+    column_id = pexConfig.Field[str](default="id", doc="Catalog index column key")
+    compute_errors = pexConfig.ChoiceField[str](
+        default="INV_HESSIAN_BESTFIT",
+        doc="Whether/how to compute sqrt(variances) of each free parameter",
+        allowed={
+            "NONE": "no errors computed",
+            "INV_HESSIAN": "inverse hessian using noisy image as data",
+            "INV_HESSIAN_BESTFIT": "inverse hessian using best-fit model as data",
+        },
+    )
+    compute_errors_from_jacobian = pexConfig.Field[bool](
+        default=True,
+        doc="Whether to estimate the Hessian from the Jacobian first, with finite differencing as a backup",
+    )
+    compute_errors_no_covar = pexConfig.Field[bool](
+        default=True,
+        doc="Whether to compute parameter errors independently, ignoring covariances",
+    )
+    config_fit = pexConfig.ConfigField[ModelFitConfig](default=ModelFitConfig, doc="Fitter configuration")
+    fit_centroid = pexConfig.Field[bool](default=True, doc="Fit centroid parameters")
+    fit_linear_init = pexConfig.Field[bool](default=True, doc="Fit linear parameters after initialization")
+    fit_linear_final = pexConfig.Field[bool](default=True, doc="Fit linear parameters after optimization")
+    flag_errors = pexConfig.DictField(
+        default={},
+        keytype=str,
+        itemtype=str,
+        doc="Flag column names to set, keyed by name of exception to catch",
+    )
+    prefix_column = pexConfig.Field[str](default="mpf_", doc="Column name prefix")
+
+    def schema(
+        self,
+        bands: list[str] = None,
```

Comment:
It's used in derived types. I'm not sure what the best way to indicate that is, other than that if you simply removed the kwarg only from here it would break every derived type.